code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -616,6 +616,10 @@
 	var/mob/living/silicon/ai/U = usr
 
 	for (var/obj/machinery/camera/C in GLOB.cameranet.cameras)
+		//cameras that are located in an object have their z set to 0, breaking the next check;
+		//mostly an issue for silicons. their cameras are in the end of the list, so it shouldn't affect building the list
+		if(!C.z)
+			continue
 		var/list/tempnetwork = C.network
 		if(!(is_station_level(C.z) || is_mining_level(C.z) || ("ss13" in tempnetwork)))
 			continue
```

Comment:
you should be able to jump to borgs. pull a turf instead.

also move the tempnetwork bit below these if's, we don't reference them in any of the checks yet so it's just wasted processing. 

```suggestion
		var/turf/camera_turf = get_turf(C)
		if(!camera_turf || !(is_station_level(camera_turf.z) || is_mining_level(camera_turf.z) || ("ss13" in tempnetwork)))
			continue
		if(!C.can_use())
			continue
			
		var/list/tempnetwork = C.network
		tempnetwork.Remove("rd", "ordnance", "prison")
```

Reply:
yeah get_turf is definitely better

jump to network does not allow you to jump to a borg camera actually. borg cameras are in ss13 network, which is a good 2/3 of all the cameras present on any station. this causes the verb to jump to what i presume was the first ss13 networked camera. i can see borg cameras being jumped to with this verb in the event that _all_ the station cameras get destroyed, which might happen in the event of a singuloose. hilarity

Reply:
also yeah we use tempnetwork in the check for ss13 network in the camera, so it kinda has to be before the ifs